### ISSUE TITLE
Product with variants without variant attributes are not added to the checkout

### ISSUE THREAD

**Issue by armandov91** [mirumee/saleor#4196]
### What I'm trying to achieve
Add a product with the selected variant and add to cart

### Steps to reproduce the problem
1. Create a product type that has no variant attributes
1. Then create a product with that type
1. Create at least two variants for it
1. Try to add this product to the checkout (any of the variants)
1. Nothing happens!

### What I expected to happen
See the selected product with the selected variant on my cart.

### My solution
I dont' know if this is the correct approach or maybe I need to update something(Maybe already fixed).

Problem occurs in : **saleor/static/js/components/checkout.js** function: `$('.product-form button').click((e)` in line **45**:
`let variant = $target.find('input[name="variant"]').val();`

When the product has only one variant it is expected not so see the only variant in a choice field as declared in **saleor/product/forms.py**
in **VariantChoiceField** line **38**:
```
# Don't display select input if there is only one variant.
        if self.queryset.count() == 1:
            self.widget = forms.HiddenInput({"value": variants.all()[0].pk})
```
			
So we get a hidden `<input>` element in our product form e.g. :
`<input type="hidden" name="variant" value="3" id="id_variant">`

This is ok when we only have one variant as our javascript will find the input element, but when we have a product with more variants,`<input>` element is replaced with a `<select>` element in order to display all the option we could have for this product but now it can not be found.(actually the input element is not been found so we get it as undefined).

What I did was :
After: `let variant = $target.find('input[name="variant"]').val();` :

`if (typeof variant === 'undefined') variant = $target.find('select[name="variant"]').children('option:selected').val();`

So basically, if our variant is undefined(not found), we search for the `<select>` element with **name="variant"**,search its children and get the selected one.

Finally, execute `npm run build-assets` again.

This succesfully add the product to the cart.

Please let me know if this is the correct approach or should be done in a different way or it is already solved.

**Comment by NyanKiyoshi** [comment]
This only happens if the variant list is the select option and not a checkbox.

Thanks for reporting! I will check this out whenever I get a few spare minutes.
